docs/contributing/contributing.md

```diff
@@ -0,0 +1,62 @@
+# Contributing to Labelmerge
+
+Labelmerge python package dependencies are managed with Poetry (`v1.2.0+`), which
+you will need installed on your machine. You can find installation instructions 
+on the [Poetry website](https://python-poetry.org/docs/master/#installation).
+
+_Note: These instructions are only recommended if you are making changes to the
+Labelmerge codebase and committing these back to the repository, or if you are 
+using Snakemake's cluster execution profiles. If not, it is easier to run 
+Labelmerge using the packaged singularity container (e.g. 
+`docker://khanlab/labelmerge:latest`)._
+
+## Setup the development environment
+
+Clone the repository and install all dependencies (including `dev`) with poetry:
+
+```
+git clone https://github.com/khanlab/labelmerge.git 
+cd labelmerge
+poetry install --with dev 
+```
+
+Poetry will automatically create a virtual environment. To customize where 
+these virtual environments are stored, see the poetry docs 
+[here](https://python-poetry.org/docs/configuration/)
+
+Then, you can run labelmerge with:
```

Comment:
This feels like this should be a separate part of the doc on how to run labelmerge rather than contribution.